### ISSUE TITLE
複数のソーシャルアカウントを紐付けたい

### ISSUE THREAD

**Issue by uchan-nos** [adventar/adventar#43]
現在 Twitter アカウントでログインして使っているのですが、同じユーザに対して GitHub などの別のソーシャルアカウントも紐付けたいです。可能でしょうか？
GitHub でログインしてみたら別のユーザとして振る舞っているように見えますので、現状は出来ないのかなと思っています。
ログインに使用しているサービスが終了する、あるいは使用を止めてアカウントを閉じたいと思ったときに困ってしまうことに気付きました。

**Comment by hokaccha** [comment]
現状だとできないですね。実装自体は難しくないのですが、既存のデータをマージする処理などが必要になって、これがけっこうたいへんなので実装に踏み切れていません。

**Comment by uchan-nos** [comment]
なるほど、データのマージが大変なんですね。例えばこういう次善策はありそうだなと思いました。
- まだ登録されていないソーシャルアカウントのみ紐付け可能とする
- 既存アカウントの削除（退会）処理を追加する

既に2つ以上のアカウントでそれぞれ記事を投稿するなどをしてる人は諦めてもらうしかないですが、「GitHub でログインしてみたら別アカウントになっちゃったなあ」という程度の人なら、まずアカウント削除してから本体のアカウントに紐付け作業することで、マージ処理を避けられると思いました。

いずれにせよ、私は要望するだけの立場ですのでhokacchaさんの決定を尊重します。

**Comment by hokaccha** [comment]
そうですね。アカウント削除すればOKとするのはよいアイデアだと思います。すぐに実装できるかは保証できませんが、優先度あげて検討してみます！

**Comment by TakamiChie** [comment]
今回のTwitter API有料化に際して認証APIが有償化対象かどうかはわかりませんが、こちらはどうなっていますか？
と、ちょっと気になったので（シーズンオフですが）コメントをしてみます。
